internal/server/iigointernal/executive.go

```diff
@@ -0,0 +1,148 @@
+package iigointernal
+
+import (
+	"github.com/SOMAS2020/SOMAS2020/internal/common/baseclient"
+	"github.com/SOMAS2020/SOMAS2020/internal/common/gamestate"
+	"github.com/SOMAS2020/SOMAS2020/internal/common/roles"
+	"github.com/SOMAS2020/SOMAS2020/internal/common/rules"
+	"github.com/SOMAS2020/SOMAS2020/internal/common/shared"
+	"github.com/SOMAS2020/SOMAS2020/internal/common/voting"
+)
+
+type executive struct {
+	ID               shared.ClientID
+	clientPresident  roles.President
+	budget           shared.Resources
+	speakerSalary    shared.Resources
+	RulesProposals   []string
+	ResourceRequests map[shared.ClientID]shared.Resources
+}
+
+// returnSpeakerSalary returns the salary to the common pool.
+func (e *executive) returnSpeakerSalary() shared.Resources {
+	x := e.speakerSalary
+	e.speakerSalary = 0
+	return x
+}
+
+// Get rule proposals to be voted on from remaining islands
+// Called by orchestration
+func (e *executive) setRuleProposals(rulesProposals []string) {
+	e.RulesProposals = rulesProposals
+}
+
+// Set approved resources request for all the remaining islands
+// Called by orchestration
+func (e *executive) setAllocationRequest(resourceRequests map[shared.ClientID]shared.Resources) {
+	e.ResourceRequests = resourceRequests
+}
+
+// Get rules to be voted on to Speaker
+// Called by orchestration at the end of the turn
+func (e *executive) getRuleForSpeaker() string {
+	e.budget -= serviceCharge
+	result, _ := e.clientPresident.PickRuleToVote(e.RulesProposals)
+	return result
+}
+
+// Send Tax map all the remaining islands
+// Called by orchestration at the end of the turn
+func (e *executive) getTaxMap(islandsResources map[shared.ClientID]shared.Resources) map[shared.ClientID]shared.Resources {
+	e.budget -= serviceCharge
+	result, _ := e.clientPresident.SetTaxationAmount(islandsResources)
+	return result
+}
+
+// broadcastTaxation broadcasts the tax amount decided by the president to all island still in the game.
+func (e *executive) broadcastTaxation(islandsResources map[shared.ClientID]shared.Resources) {
+	e.budget -= serviceCharge
+	taxAmountMap := e.getTaxMap(islandsResources)
```

Comment:
So the president always has to pay two service charges here? shouldnt they just be charged 2xservice charge here for simplicity

Reply:
For MVP, we costed actions by having a flat rate for every function call because we had no idea what a reasonable cost was. I can see that this seems silly but the way we deduct costs is going to be overhauled post-MVP anyhow so if it's alright with you, I'm not going to make this change as it isn't an issue per se. 

Also, I think it would be more worthwhile to change this once #141 is resolved by design.